download.js

```diff
@@ -1,40 +1,58 @@
-const fs = require('fs')
-const publicKey = process.env.ONE_SKY_PUBLIC_KEY
-const privateKey = process.env.ONE_SKY_PRIVATE_KEY
-
-const onesky = require('onesky')(publicKey, privateKey)
+const { promises: fs } = require('fs')
+const oneSky = require('@brainly/onesky-utils')
 const sanitizeHtml = require('sanitize-html')
-const sanitizeOpts = {
-  allowedTags: ['b', 'strong', 'a', 'code'],
-  allowedAttributes: {
-    a: ['href', 'class']
-  },
-  textFilter(text) {
-    return text.replace(/\{\{/, '&#123;&#123;').replace(/\}\}/, '&#125;&#125;')
-  }
-}
 
-onesky.string.output({ platformId: '25049' }, function(err, r) {
-  if (err) {
-    console.error(err)
-    return
-  }
+async function run() {
+  try {
+    // The recommended OneSky set-up appears to require an API request to
+    // generate files on their side, which you could then request and use. We
+    // only have 1 such file that appears to be misnamed (en-US, despite our
+    // translations being marked as GB) and very out-of-date.
+    // However by requesting the "multilingual file" for this file, we get all
+    // of the translations
+    const content = await oneSky.getMultilingualFile({
+      apiKey: process.env.ONE_SKY_PUBLIC_KEY,
+      secret: process.env.ONE_SKY_PRIVATE_KEY,
+      projectId: '25049',
+      fileName: 'en-US.json'
+    })
+    const json = JSON.parse(content)
 
-  for (const [code, lang] of Object.entries(r.translation['en-US.json'])) {
-    for (const [key, value] of Object.entries(lang)) {
-      lang[key] = sanitizeHtml(value, sanitizeOpts)
+    for (const [code, lang] of Object.entries(json)) {
+      for (let [key, value] of Object.entries(lang.translation)) {
+        // Handle multi-line strings as arrays by joining on newline
+        if (Array.isArray(value)) {
+          value = value.join('\n')
+        }
+        lang.translation[key] = sanitize(value)
+      }
+
+      await fs.writeFile(
+        `./locales/${code}.json`,
+        JSON.stringify(lang.translation, null, 2)
+      )
     }
 
-    fs.writeFileSync(`./locales/${code}.json`, JSON.stringify(lang, null, 2))
+    // Copy files, so we have appropriate dialects
+    await fs.copyFile('./locales/en-GB.json', './locales/en-US.json')
+    await fs.copyFile('./locales/en-GB.json', './locales/en.json')
+    await fs.copyFile('./locales/zh-CN.json', './locales/cn.json')
+  } catch (error) {
+    console.error(error)
```

Comment:
Let's bail-out with a non-zero exit code here. The CI job should fail when the download script is not working.

Reply:
I think just not catching the error would work too?

Reply:
> I think just not catching the error would work too?



It does not seem so, node will just log a `UnhandledPromiseRejectionWarning` and exit with code 0.

<details>

```js
async function work() {
  throw new Error('hi there')
}

async function main() {
  await work()
}

main()
```

```
/tmp$ node demo.js 
(node:21407) UnhandledPromiseRejectionWarning: Error: hi there
    at work (/tmp/demo.js:2:9)
    at main (/tmp/demo.js:6:9)
    at Object.<anonymous> (/tmp/demo.js:9:1)
    at Module._compile (internal/modules/cjs/loader.js:1158:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1178:10)
    at Module.load (internal/modules/cjs/loader.js:1002:32)
    at Function.Module._load (internal/modules/cjs/loader.js:901:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47
(node:21407) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:21407) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
/tmp$ echo $?
0
```

(Note that the hinted cli flag works, but I prefer to make it obvious from the script that we expect an error and handle it.)

``` 
/tmp$ node --unhandled-rejections=strict demo.js 
/tmp/demo.js:2
  throw new Error('hi there')
        ^

Error: hi there
    at work (/tmp/demo.js:2:9)
    at main (/tmp/demo.js:6:9)
    at Object.<anonymous> (/tmp/demo.js:9:1)
    at Module._compile (internal/modules/cjs/loader.js:1158:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1178:10)
    at Module.load (internal/modules/cjs/loader.js:1002:32)
    at Function.Module._load (internal/modules/cjs/loader.js:901:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47
/tmp$ echo $?
1
``` 
</details>

Reply:
Thanks for checking, I'll exit the process then.